### ISSUE TITLE
public methods for getting trustStore, keyStore, KeyManager, TrustManager out of the SSLFactory

### ISSUE THREAD

**Issue by HoffiMuc** [Hakky54/sslcontext-kickstart#132]
Hi, nice project ... I like it.

My request would be:

once I have created and "loaded" everything I want into the `val sslFactory: SSLFactory`

it would be nice if I could pass this around and _still_ be able to also get the finer building blocks of it.

I'd love to have public(!) methods on it (or somewhere else) to get its

- trustStore (KeyStore)
- keyStore (KeyStore)
- X509ExtendedKeyManager
- X509ExtendedTrustManager
- etc.

maybe it is already possible, and I just didn't find on where to find it "in the deeper API"

thanx for considering!

**Comment by Hakky54** [comment]
Hi @HoffiMuc 

Thank you for your kind words! 

I get your feature request and I actually need to say that this was available before but I removed from version 7.0.0 onwards. In the early stage of development of this library it made sense to hold on these individual building blocks such as the keystore, truststore, keymanager and trustmanager within the sslfactory. Those objects could be fetched later on.

However the library evolved and got new capabilities such as building the sslfactory from a list of certifficates or a private key object. The most challenging decision was adding a hot swappable keymanager and trustmanager to the sslfactory which made the option of storing the initial keystores, truststore, keymanager and trustmanager useless... So I ended up removing those options.

You can still get the keymanager and trustmanager from the sslfactory, but not the individual ones if you are using multiple because the sslfactory will return the composed one.

So a workaround would be maybe using the following utils within the library:

- [CertificateUtils](https://github.com/Hakky54/sslcontext-kickstart/blob/master/sslcontext-kickstart/src/main/java/nl/altindag/ssl/util/CertificateUtils.java)
- [KeyStoreUtils](https://github.com/Hakky54/sslcontext-kickstart/blob/master/sslcontext-kickstart/src/main/java/nl/altindag/ssl/util/KeyStoreUtils.java)
- [KeyManagerUtils](https://github.com/Hakky54/sslcontext-kickstart/blob/master/sslcontext-kickstart/src/main/java/nl/altindag/ssl/util/KeyManagerUtils.java)
- [TrustManagerUtils](TrustManagerUtils)

These utils are used within the sslfactory and it will return the objects you want to create such as the keystore, truststore, keymanager and trustmanager. The downside would be that you need to store it in your custom data class. But the positive side is you can use the utility classes which give more advanced control in how you want to create and manage the objects and it also takes away the verbosity and validations.

Would that workaround work for you?

So basically what you can do it:
```
KeyStore identityOne = KeyStoreUtils.loadKeyStore(identityStorePathOne, identityStorePassword, identityStoreType);
KeyStore identityTwo = KeyStoreUtils.loadKeyStore(identityStorePathTwo, identityStorePassword, identityStoreType);
KeyStore trustStoreOne = KeyStoreUtils.loadKeyStore(trustStorePathTwo, trustStorePassword, trustStoreType);
KeyStore trustStoreTwo = KeyStoreUtils.loadKeyStore(trustStorePathTwo, trustStorePassword, trustStoreType);

SSLMaterial sslMaterial = new SSLMaterial(
    Arrays.asList(identityOne, identityTwo),
    Arrays.asList(trustStoreOne, trustStoreTwo)
);
```
